src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -22,14 +23,38 @@ sealed class BlobFeedAction
 
         public BlobFeed feed;
         public int MaxClients { get; set; } = 8;
+        
+        const string feedRegex = @"https:\/\/(.*?)\.blob\.core\.windows\.net\/(.*?)\/(.*)\/index\.json";
```

Comment:
These regex's scare me a little as I expect they are going to change over time.  How confident are we that the URL's aren't going to change? Do we think it is worth making these parameters to the task so we can change them without requiring a tool update if they do change?

Reply:
Can you please combine these into one with the following regex and update to using ispublic instead of securepath.? 

```
            string feedRegex = @"(?<feedurl>https:\/\/(?<accountname>[^\.-]+)(?<domain>[^\/]*)\/((?<token>[a-zA-Z0-9+\/]*?\/\d{4}-\d{2}-\d{2})\/)?(?<containername>[^\/]+)\/(?<relativepath>.*)\/)index\.json";

            Match m = Regex.Match(expectedUrl, feedRegex);

            if (m.Success)
            {
                string accountName = m.Groups["accountname"].Value;
                string containerName = m.Groups["containername"].Value;
                string relativePath = m.Groups["relativepath"].Value;
                string feedUrl = m.Groups["feedurl"].Value;

                bool ispublic = string.IsNullOrWhiteSpace(m.Groups["token"].Value);
                this.feed = new BlobFeed(accountName, accountKey, containerName, feedUrl, string.IsNullOrWhiteSpace(indexDirectory) ? Path.GetTempPath() : indexDirectory, Log, isPublic)
            }
```

With that you should minimize any assumptions about the url pieces and only depend on the format.